### PR TITLE
HcalDQM: Add protection to collection non-exist case for GPUTask

### PR DESCRIPTION
**Pull request by lwang046**
#### PR description:
Regarding issue #40115, this PR corrects the reference&target name mixing in the hcalgpu task, adds some conditions to the collections non-exist case instead of throwing out fetal exception, and correct a minor numeric error in the DigiTask that should mod the LS by 60, instead of 50. 

#### PR validation:
Tested with `cmsRun DQM/Integration/python/clients/hcalgpu_dqm_sourceclient-live_cfg.py runInputDir=/afs/cern.ch/work/l/lowang/DQM_Dev/CMSSW_12_6_X_2022-11-17-2300/src/ runNumber=362217 runkey=hi_run` and shows no problems.

### CODE REVIEW

File: DQM/HcalTasks/plugins/HcalGPUComparisonTask.cc [hunk 1]
```diff
@@ -96,10 +96,40 @@ HcalGPUComparisonTask::HcalGPUComparisonTask(edm::ParameterSet const& ps)
...
+      HcalDetId did = it->id();
+      energyGPUvsCPU_subdet_.fill(did, -0.5, energy);
+    }
+    return;
+  }
+  if (!gotRefCollection && !gotTargetCollection) {
+    edm::LogWarning("HcalGPUComparisonTask")
+        << "Neither CPU nor GPU RecHit Collection available, will not fill this event." << std::endl;
+    return;
+  }
```
Comment (missirol) [inline DQM/HcalTasks/plugins/HcalGPUComparisonTask.cc:132]: Let me suggest an optional change to make this more compact; I did not test this, so please double-check this if you want to use it.
```cpp
  auto const chbhe_ref = e.getHandle(tokHBHE_ref_);
  auto const chbhe_target = e.getHandle(tokHBHE_target_);

  if (not (chbhe_ref.isValid() and chbhe_target.isValid())) {

    if (chbhe_target.isValid()) {
      edm::LogWarning("HcalGPUComparisonTask")
          << "The CPU HBHERecHitCollection " << tagHBHE_ref_.encode() << " is not available";

      for (auto const& rh : *chbhe_target)
        energyGPUvsCPU_subdet_.fill(rh.id(), -0.5, rh.energy());
    }
    else if (chbhe_ref.isValid()) {
      edm::LogWarning("HcalGPUComparisonTask")
          << "The GPU HBHERecHitCollection " << tagHBHE_target_.encode() << " is not available";

      for (auto const& rh : *chbhe_ref)
        energyGPUvsCPU_subdet_.fill(rh.id(), rh.energy(), -0.5);
    }
    else {
      edm::LogWarning("HcalGPUComparisonTask")
          << "Neither CPU nor GPU RecHit Collection available, will not fill this event.";
    }

    return;
  }
```

File: DQM/HcalTasks/plugins/HcalGPUComparisonTask.cc [hunk 2]
```diff
@@ -115,7 +145,8 @@ HcalGPUComparisonTask::HcalGPUComparisonTask(edm::ParameterSet const& ps)
     if (mRecHitEnergy.find(did) == mRecHitEnergy.end())
       mRecHitEnergy.insert(std::make_pair(did, energy));
     else
-      edm::LogError("HcalDQM|RechitTask") << "Duplicate Rechit from the same HcalDetId";
+      edm::LogError("HcalGPUComparisonTask") << "Duplicate Rechit from the same HcalDetId" << std::endl;
+    ;
```
Comment (missirol) [inline DQM/HcalTasks/plugins/HcalGPUComparisonTask.cc:149]: `std::endl` here is not needed, as `Log*` calls ignore it and apply a newline regardless.

**Comment by emanueleusai** [comment]
type hcal,bugfix

**Comment by emanueleusai** [comment]
please test

**Comment by perrotta** [comment]
@syuvivida is there any reason why you signed the [backport of this PR](https://github.com/cms-sw/cmssw/pull/40118), and not this one in the master?

**Comment by syuvivida** [comment]
+1

**Comment by rvenditti** [comment]
please test

**Comment by rvenditti** [comment]
please test

**Comment by emanueleusai** [comment]
apologies commented in the wrong PR

**Comment by emanueleusai** [comment]
+1

**Comment by perrotta** [comment]
+1

**Pull request merged**